test/new-e2e/examples/customenv_with_two_vm_test.go

```diff
@@ -6,49 +6,195 @@
+func multiVMEnvProvisionerTwo() e2e.PulumiEnvRunFunc[multiVMEnv] {
 	return func(ctx *pulumi.Context, env *multiVMEnv) error {
+		// first, let's create an aws environment
+		// this is where our remote hosts will live
 		awsEnv, err := aws.NewEnvironment(ctx)
+		if err != nil {
+			return err
+		}
 
-		main, err := ec2.NewVM(awsEnv, "main", ec2.WithOS(os.UbuntuDefault))
+		// We create a first VM where we install a workload application using docker
+		// We use os.AmazonLinuxECSDefault to have linux installed with docker
+		appHost, err := awshost.NewHost(awshost.AwsHostParams{
+			Name:           "app",
+			PulumiContext:  ctx,
+			AwsEnvironment: &awsEnv,
+			Options:        []ec2.VMOption{ec2.WithOS(compos.AmazonLinuxECSDefault)},
+			Importer:       &env.AppVM.HostOutput,
+		})
 		if err != nil {
 			return err
 		}
-		main.Export(ctx, &env.MainVM.HostOutput)
 
-		app, err := ec2.NewVM(awsEnv, "app", ec2.WithOS(os.AmazonLinuxDefault))
+		// initialize a docker manager without installing docker.io (that false)
+		dockerManager, _, err := docker.NewManager(*awsEnv.CommonEnvironment, appHost, false)
+		if err != nil {
+			return err
+		}
+		// we docker compose the httpbin application
+		_, err = dockerManager.ComposeStrUp("httpbin", []docker.ComposeInlineManifest{{
+			Name:    "httpbin",
+			Content: pulumi.String(httpbinComposeContent),
+		}}, pulumi.StringMap{})
+		if err != nil {
+			return err
+		}
+
+		// we deploy a fakeintake
+		_, err = awsecs.NewFakeintake(awsecs.FakeintakeParams{
+			Name:           "fakeintake",
+			PulumiContext:  ctx,
+			AwsEnvironment: &awsEnv,
+			Importer:       &env.FakeIntake.FakeintakeOutput,
+		})
+		if err != nil {
+			return err
+		}
+
+		// Finally we create a second VM where we install the agent
+		mainHost, err := ec2.NewVM(awsEnv, "main")
+		if err != nil {
+			return err
+		}
+		mainHost.Export(ctx, &env.MainVM.HostOutput)
+		// we install the agent on host
+		_, err = agent.NewHostAgent(awsEnv.CommonEnvironment, mainHost)
 		if err != nil {
 			return err
 		}
-		app.Export(ctx, &env.AppVM.HostOutput)
 
 		return nil
 	}
 }
 
+// Init is called by the suite to initialize the environment
+// once pulumi is done provisioning the resources
+// This is where we initialise clients
+func (e *multiVMEnv) Init(ctx e2e.Context) error {
+	privateKeyPath, err := runner.GetProfile().ParamStore().GetWithDefault(parameters.PrivateKeyPath, "")
+	if err != nil {
+		return err
+	}
+
+	e.AppDocker, err = client.NewDocker(ctx.T(), e.AppVM.HostOutput, privateKeyPath)
+	if err != nil {
+		return err
+	}
+
+	agent, err := client.NewHostAgentClient(ctx.T(), e.MainVM, true)
+	if err != nil {
+		return err
+	}
+	e.Agent = &agent
+	return nil
+}
+
 type multiVMSuite struct {
 	e2e.BaseSuite[multiVMEnv]
 }
 
 func TestMultiVMSuite(t *testing.T) {
-	e2e.Run(t, &multiVMSuite{}, e2e.WithPulumiProvisioner(multiVMEnvProvisioner(), nil))
+	runFunc := multiVMEnvProvisionerOne
+	if os.Getenv("USE_VARIANT_TWO") != "" {
```

Comment:
not sure this is the best way to run different env, I would prefer to have separated files as today running the same common tests (as today)